src/main/scala/rise/elevate/rules/algorithmic.scala

```diff
@@ -58,7 +58,7 @@ object algorithmic {
 
   // padEmpty n >> padEmpty m -> padEmpty n + m
   @rule def padEmptyFusion: Strategy[Rise] = {
-    case e @ App(DepApp(NatKind, padEmpty(), m: Nat), App(DepApp(NatKind, padEmpty(), n: Nat), in)) =>
+    case e @ App(DepApp(_, padEmpty(), m: Nat), App(DepApp(_, padEmpty(), n: Nat), in)) =>
```

Comment:
Same question here: why did you drop the explicit kinds in the matching everywhere?

Reply:
Ah, I should roll back this too. On my first approach the `I` parameter to `Kind` was not covariant (since `idName`, and `toIdentifier` were methods in `Kind` instead of functions outside of it), and Scala complained about matching on the `kind`. I believe this should work now though, I will try to roll back.

Reply:
Restored